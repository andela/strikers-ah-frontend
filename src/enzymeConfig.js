import Enzyme from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import 'jest-localstorage-mock';

export default Enzyme.configure({
  adapter: new Adapter(),
});
