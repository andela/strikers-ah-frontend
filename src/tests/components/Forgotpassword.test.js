/* eslint-disable react/jsx-filename-extension */
import React from 'react';
import { shallow } from 'enzyme';
import '../../enzymeConfig';
import {
  ForgotPassword,
  mapStateToProps,
} from '../../components/ForgotPassword';

let props = {
  handleSubmit: jest.fn(),
  forgotPassword: jest.fn(),
  handleChange: jest.fn(),
  forgetPasswordState: {
    response: {
      message: 'something',
    },
  },
  mapStateToProps: jest.fn(),
};
let wrapper = shallow(<ForgotPassword {...props} />);
describe('`ForgotPassword.jsx`', () => {
  test('should render', () => {
    expect(wrapper).toMatchSnapshot();
  });

  test('should `click the button`', () => {
    const spy = jest.spyOn(wrapper.instance(), 'handleSubmit');
    wrapper.instance().forceUpdate();
    const fakeEvent = { preventDefault: () => {} };

    const IsForm = wrapper.find('#forgotPasswordForm');
    IsForm.simulate('submit', fakeEvent);
    expect(IsForm.length).toBe(1);
    expect(spy).toHaveBeenCalled();
  });
  props = {
    handleSubmit: jest.fn(),
    forgotPassword: jest.fn(),
    handleChange: jest.fn(),
    mapStateToProps: jest.fn(),
  };
  wrapper = shallow(<ForgotPassword {...props} />);
  test('should `click the button`', () => {
    const spy = jest.spyOn(wrapper.instance(), 'handleChange');
    wrapper.instance().forceUpdate();
    const fakeEmail = {
      target: { value: 'email@f.com' },
    };

    const IsForm = wrapper.find('InputForm');
    IsForm.simulate('change', fakeEmail);
    expect(IsForm.length).toBe(1);
    expect(spy).toHaveBeenCalled();
  });
  test('should call the function ', () => {
    const state = {
      rateArticle: {
        rate: {
          error: '',
        },
      },
    };
    mapStateToProps(state);
  });
});
