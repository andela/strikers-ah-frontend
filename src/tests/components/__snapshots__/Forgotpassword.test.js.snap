// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`\`ForgotPassword.jsx\` should render 1`] = `ShallowWrapper {}`;
